public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,833 @@
+      configs,
+    });
+    let resp = _.get(response, 'data.response');
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    if (ok) {
+      context.setFlyout(null);
+      context.renderFlyout(configs, resp, context.queriesForOverview);
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
+    throw 'There was a problem validating the configurations';
+  }
+}
+
+const ConfigCell = (props) => {
+  return (
+    <FixedWidthRow label={props.title}>
+      <EuiText>
+        <p className="enabled">{props.description}</p>
+      </EuiText>
+    </FixedWidthRow>
+  );
+};
+
+const validationErrorCallOut = (failures, suggestedChanges, field) => {
+  let message;
+  for (let [key, value] of Object.entries(failures)) {
+    if (key === 'duplicates' && field === 'name') {
+      message = 'Detector name is a duplicate';
+    } else if (key === 'missing' && value[0] === field) {
+      message = 'This field is required';
+    } else if ((key === 'regex' && field === 'name') || (key === 'format' && field === 'name')) {
+      message = 'Valid characters are a-z, A-Z, 0-9, -(hyphen) and _(underscore)';
+    }
+  }
+  for (let [key, value] of Object.entries(suggestedChanges)) {
+    if (key === 'window_delay' && field === 'window_delay') {
+      message = 'Window delay should be at least ' + value[0] + ' minutes';
+    } else if (key === 'filter_query' && field === 'filter_query') {
+      message = value[0];
+    } else if (key === 'detectionIntervalMax' && field === 'detection_interval') {
+      message = value;
+    }
+  }
+  if (message) {
+    return (
+      <EuiCallOut
+        style={{ paddingBottom: '2px' }}
+        title={message}
+        iconType="alert"
+        size="s"
+        color="danger"
+      ></EuiCallOut>
+    );
+  }
+  return null;
+};
+
+const createDetector = (context) => {
+  const [isFilterPopoverOpen, setFilterPopoverOpen] = useState(false);
+  const [dataTypes, setDataTypes] = useState({});
+
+  useEffect(() => {
+    onQueryMappings();
+  }, []);
+
+  const handleFieldChange = (option, field, form) => {
+    form.setFieldValue(field.name, option);
+    // User can remove where condition
+    if (option.length === 0) {
+      form.setFieldError('where', undefined);
+    }
+  };
+
+  async function onQueryMappings() {
+    const index = context.adConfigs.indices;
+    try {
+      const mappings = await queryMappings(index);
+      const data = getPathsPerDataType(mappings);
+      setDataTypes(data);
+    } catch (err) {
+      console.error('There was an error getting mappings for query', err);
+    }
+  }
+  const handleChangeWrapper = (e, field) => {
+    field.onChange(e);
+  };
+
+  const helpTextInterval = (context) => {
+    if (
+      Object.keys(context.failures).length != 0 &&
+      toString(context.adConfigs.detection_interval) == '1'
+    ) {
+      return "*Detector interval hasn't been validated yet, please fix other failures first";
```

Comment:
what is the reason for the `*` here? Is it converted to a bullet point in markdown?